static/panes/pane.ts

```diff
@@ -139,22 +140,6 @@ export abstract class Pane<E extends monaco.editor.IEditor, S> {
      */
     registerEditorActions(): void {}
 
-    /**
```

Comment:
I think I'm lost here -- I will try and look at this tomorrow. I've not been keeping track of all the pane-name changes and I'm having a hard time following the thread of this. My IDE is pretty upset about missing things (e.g. `Compiler.prototype.updateTitle` seems to be accessing an unknown field `this.paneName`...) and this `getPaneName()` seemed to be part of an effort to de-duplicate code, but now I'm not sure why you had to remove it?

Reply:
I am not sure I made all the best design decisions, but I gave it a stab. My suggested change with getPaneName wasn't to remove it but change it from being a function only returning a name like "LLVM Output (Editor #1, Compiler #1)" to be a function that returns the actual name of the pane, i.e. the custom name or the default "LLVM Output (Editor #1, Compiler #1)"-style name. Currently in the code base, uses of `getPaneName` have to do check for a custom name every time `this.paneName ? this.paneName : this.getPaneName()`.

The proposed data flow looks like
`getPaneName()` returns either a custom `this.paneName` or a default name
`getDefaultPaneName()` returns the default name, e.g. "LLVM Output", the idea was this would be the only method a pane would have to override
`getPaneTag()` returns the "(Editor #1, Compiler #1)" string

Reply:
```js
this.paneName ? this.paneName : this.getPaneName()
```

If I understand right you're dynamically testing for whether "this.paneName" exists at all? That's a little sublte and murky (which is why the IDE highlights it as a probable mistake).

I think it's considered bad form to dynamically add members in this way: this is also a problem when `this.paneName = "";` is equivalent to `this.paneName` was never defined at all (which may be intentional, but again is subtle). I'd prefer I think explicitly initialising it to `null` (or `""` if that makes sense) in the constructor to say "this is at least expected to be here".

Happy to chat more - sorry for the stream of consciousness review...I'm exposing my own limited understanding of the code these days and need to get myself re-acquainted in order to make more informed reviews!

Reply:
Again, sorry for jumping the gun on this; I was definitely speaking out of turn without understanding how the code had evolved. I think the PR I have brewing covers my concerns, so if nobody else minds I think this is great as-is. I've asked a more TS-savvy person to do a quick review of the rest of this PR.

Reply:
All good!